File: backend/src/schemas/conversation.py
"""Conversation message schemas."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class MessageCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=50_000)

    model_config = ConfigDict(extra="forbid")


class MessageResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    role: str  # "user" | "assistant"
    content: str
    request_id: uuid.UUID | None
    actions: list
    source: str
    external_id: str | None
    thread_ref: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SendMessageResponse(BaseModel):
    """Returned after a user message is classified + extracted."""

    message: MessageResponse
    intent: str  # "chit_chat" | "question" | "request" | "modification"
    request_id: uuid.UUID | None
    request_created: bool
    intent_summary: str | None = None

File: frontend/src/api/conversation.ts
import apiClient from './client'

export interface Message {
  id: string
  project_id: string
  role: 'user' | 'assistant'
  content: string
  request_id: string | null
  actions: unknown[]
  source: string
  external_id: string | null
  thread_ref: string | null
  created_at: string
}

export type MessageIntent = 'chit_chat' | 'question' | 'request' | 'modification'

export interface SendMessageResponse {
  message: Message
  intent: MessageIntent
  request_id: string | null
  request_created: boolean
  intent_summary: string | null
}

export const conversationApi = {
  list: async (projectId: string): Promise<Message[]> => {
    const { data } = await apiClient.get<Message[]>(
      `/api/v1/projects/${projectId}/messages`,
    )
    return data
  },
  send: async (projectId: string, content: string): Promise<SendMessageResponse> => {
    const { data } = await apiClient.post<SendMessageResponse>(
      `/api/v1/projects/${projectId}/messages`,
      { content },
    )
    return data
  },
}
